Loaders/Beebyte-Deobfuscator/Extensions.cs

```diff
@@ -0,0 +1,151 @@
+        }
+        public static LookupType ToLookupType(this TypeInfo type, LookupModel lookupModel)
+        {
+            if (type == null) return new LookupType(lookupModel) { };
+            if (lookupModel.ProcessedIl2CppTypes.Contains(type))
+            {
+                return lookupModel.Il2CppTypeMatches[type];
+            }
+            lookupModel.ProcessedIl2CppTypes.Add(type);
+            LookupType t = new LookupType(lookupModel) { Il2CppType = type, Children = new List<LookupType>() };
+            lookupModel.Il2CppTypeMatches.Add(type, t);
+            lookupModel.Il2CppTypeMatches[type].Fields = type.DeclaredFields.ToLookupFieldList(lookupModel).ToList();
+            lookupModel.Il2CppTypeMatches[type].DeclaringType = type.DeclaringType.ToLookupType(lookupModel);
+            lookupModel.Il2CppTypeMatches[type].Properties = type.DeclaredProperties.ToLookupPropertyList(lookupModel).ToList();
+            lookupModel.Il2CppTypeMatches[type].Methods = type.DeclaredMethods.ToLookupMethodList(lookupModel).ToList();
+
+            if (!lookupModel.Il2CppTypeMatches[type].DeclaringType.IsEmpty)
+            {
+                lookupModel.Il2CppTypeMatches[type].DeclaringType.Children.Add(lookupModel.Il2CppTypeMatches[type]);
+            }
+            return lookupModel.Il2CppTypeMatches[type];
+        }
+        public static LookupType ToLookupType(this TypeDef type, LookupModel lookupModel)
+        {
+            if (type == null) return new LookupType(lookupModel);
+            if (lookupModel.ProcessedMonoTypes.Contains(type))
+            {
+                return lookupModel.MonoTypeMatches[type];
+            }
+            lookupModel.ProcessedMonoTypes.Add(type);
+            LookupType t = new LookupType(lookupModel) { MonoType = type, Children = new List<LookupType>() };
+            lookupModel.MonoTypeMatches.Add(type, t);
+            lookupModel.MonoTypeMatches[type].Fields = type.Fields.ToLookupFieldList(lookupModel).ToList();
+            lookupModel.MonoTypeMatches[type].DeclaringType = type.DeclaringType.ToLookupType(lookupModel);
+            lookupModel.MonoTypeMatches[type].Properties = type.Properties.ToLookupPropertyList(lookupModel).ToList();
+            lookupModel.MonoTypeMatches[type].Methods = type.Methods.ToLookupMethodList(lookupModel).ToList();
+
+            if (!lookupModel.MonoTypeMatches[type].DeclaringType.IsEmpty)
+            {
+                lookupModel.MonoTypeMatches[type].DeclaringType.Children.Add(lookupModel.MonoTypeMatches[type]);
+            }
+            return lookupModel.MonoTypeMatches[type];
+        }
+        public static IEnumerable<LookupType> ToLookupTypeList(this IEnumerable<TypeDef> monoTypes, LookupModel lookupModel)
+        {
+            foreach (TypeDef type in monoTypes)
+            {
+                if (type.Name.Contains("Module")) continue;
+                yield return type.ToLookupType(lookupModel);
```

Comment:
This line uses 66% of the entire processing time, because it generates types recursively, with ToLookupType calling itself. This top-level foreach will re-iterate every type in every nested type leading to a superlinear increase in execution time on repeated types. Probably add in a check for !IsNested or similar to resolve this. Also needs to be handled for all other calls to ToLookupType in the plugin.